packages/wonder-blocks-button/components/button-core.js

```diff
@@ -0,0 +1,271 @@
+            default: {
+                background: "none",
+                color: color,
+                borderColor: Color.offBlack50,
+                borderStyle: "solid",
+                borderWidth: 1,
+            },
+            focus: {
+                background: Color.white,
+                borderColor: color,
+                borderWidth: 2,
+                paddingLeft: 14,
+                paddingRight: 14,
+            },
+            active: {
+                background: mix(fade(color, 0.32), Color.white),
+                color: mix(Color.offBlack32, color),
+                borderColor: mix(Color.offBlack32, color),
+            },
+            disabled: {
+                color: Color.offBlack32,
+                borderColor: Color.offBlack32,
+            },
+        };
+    } else if (kind === "secondary" && light) {
+        newStyles = {
+            default: {
+                background: "none",
+                color: Color.white,
+                borderColor: Color.white64,
+                borderStyle: "solid",
+                borderWidth: 1,
+            },
+            focus: {
+                color: Color.white,
+                borderColor: Color.white,
+                borderWidth: 2,
+                paddingLeft: 14,
+                paddingRight: 14,
+            },
+            active: {
+                background: mix(Color.offBlack32, color),
+                color: mix(fade(color, 0.32), Color.white),
+                borderColor: mix(fade(color, 0.32), Color.white),
+            },
+            disabled: {
+                color: mix(Color.white, fade(color, 0.32)),
+                borderColor: mix(Color.white, fade(color, 0.32)),
+            },
+        };
+    } else if (kind === "tertiary" && !light) {
+        newStyles = {
+            default: {
+                background: "none",
+                color: color,
+                paddingLeft: 4,
+                paddingRight: 4,
+            },
+            focus: {
+                borderColor: color,
+                borderStyle: "solid",
+                borderWidth: 2,
+            },
+            active: {
+                color: mix(Color.offBlack32, color),
+            },
+            disabled: {
+                color: Color.offBlack32,
+            },
+        };
+    } else if (kind === "tertiary" && light) {
+        newStyles = {
+            default: {
+                background: "none",
+                color: Color.white,
+                paddingLeft: 4,
+                paddingRight: 4,
+            },
+            focus: {
+                color: Color.white,
+                borderColor: Color.white,
+                borderStyle: "solid",
+                borderWidth: 2,
+            },
+            active: {
+                color: mix(fade(color, 0.32), Color.white),
+            },
+            disabled: {
+                color: mix(Color.white, fade(color, 0.32)),
+            },
+        };
+    }
```

Comment:
It feels like we should probably throw an exception if none of these options are selected - it would be a really bad sign if we got to that point!